frontend/src/utils/index.ts

```diff
@@ -234,12 +231,155 @@ function makeMapKey(type: 'zone' | 'fsp' | 'folder', name: string): string {
   return `${type}_${name}`;
 }
 
+/**
+ * Constructs an API URL with proper encoding for path segments and query parameters.
+ * This is the recommended way to build URLs for use with sendFetchRequest.
+ *
+ * @param basePath - The base API path (e.g., '/api/files/')
+ * @param pathSegments - Optional path segments to append (e.g., fspName)
+ * @param queryParams - Optional query parameters as key-value pairs
+ * @returns A properly encoded URL string ready for sendFetchRequest
+ *
+ * @example
+ * // Simple path
+ * buildApiUrl('/api/files/', ['myFSP'])
+ * // Returns: '/api/files/myFSP'
+ *
+ * @example
+ * // Path with query parameters
+ * buildApiUrl('/api/files/', ['myFSP'], { subpath: 'folder/file.txt' })
+ * // Returns: '/api/files/myFSP?subpath=folder%2Ffile.txt'
+ *
+ * @example
+ * // Multiple query parameters
+ * buildApiUrl('/api/ticket', [], { fsp_name: 'myFSP', path: 'folder/file.txt' })
+ * // Returns: '/api/ticket?fsp_name=myFSP&path=folder%2Ffile.txt'
+ */
+function buildApiUrl(
```

Comment:
buildApiUrl, buildExternalUrlWithQuery & buildExternalUrlWithPath all look very similar. Could these be combined into a single function with different arguments? It would be less of a mental load when trying to remember which function name to call.

Reply:
This looks much better to me. I would go with it.